backend/protocol_rpc/endpoints.py

```diff
@@ -319,20 +319,14 @@ def send_raw_transaction(
     transaction_data = {}
     result = {}
     transaction_type = -1
-    if to_address and to_address != "0x":
-        # Contract Call
-        if not accounts_manager.is_valid_address(to_address):
-            raise InvalidAddressError(
-                to_address, f"Invalid address to_address: {to_address}"
-            )
-        decoded_data = decode_method_call_data(decoded_transaction.data)
-        transaction_data = {
-            "function_name": decoded_data.function_name,
-            "function_args": decoded_data.function_args,
-        }
-        transaction_type = 2
-    else:
+    if not decoded_transaction.data:
+        # Sending value transaction
+        transaction_type = 0
+    elif not to_address or to_address == "0x":
         # Contract deployment
+        if value > 0:
+            raise InvalidTransactionError("Deploy Transaction can't send value")
```

Comment:
What do you think about moving this logic to `decode_signed_transaction`? I think that `decode_signed_transaction` should return a valid transaction, thus raising this error if it's not valid

Reply:
The transaction data is going to be saved encoded and decoded in the node, see #463